### DISCUSSION TITLE
Unable to Import database

### DISCUSSION THREAD

**Discussion by Shailesh913** [osm-search/Nominatim#3533] [General]
Hello All ,

I am facing issue while trying to import I m facing below error.

root@4bf331d26bf3:/app# PGPASSWORD=your_password PGUSER=postgres nominatim import --osm-file /app/data/berlin-latest.osm.pbf
2024-09-10 08:28:55: Using project directory: /app
2024-09-10 08:28:56: Creating database
createdb: error: could not connect to database template1: FATAL:  Peer authentication failed for user "postgres"
2024-09-10 08:28:56: FATAL: Creating new database failed.
root@4bf331d26bf3:/app#

Tried many different approaches like Postgresql Remote setup , Container but still facing error while importing.

Any help is highly appreciated . Stuck from Long time . Kindly help.

Thanks in advance !!!

**Comment by mtmail** [comment]
Can you add which version of Nominatim you're using and which installation documentation you followed? There are some different of the current version vs previous versions how database access is configured. Is this a root server, virtual machine, docker container? If docker how did you start the container?

**Reply by Shailesh913** [reply]
Thank you for your response !!
Nominatim version - 4.4.1
the output is from docker container.
I started the container by pulling the ubuntu image and then exec -it command .


also I tried with Nominatim-env (i.e. virtual environment)
(nominatim-env) root@debian:~# nominatim --version
Nominatim version 4.4.1-0 (896a40b7)
(nominatim-env) root@debian:~# 

still getting same error.

(nominatim-env) root@debian:~# sudo -u postgres psql
psql (16.4 (Debian 16.4-1.pgdg120+1), server 14.13 (Debian 14.13-1.pgdg120+1))
Type "help" for help.

postgres=# \du
                             List of roles
 Role name |                         Attributes                         
-----------+------------------------------------------------------------
 nominatim | Superuser, Create role, Create DB
 postgres  | Superuser, Create role, Create DB, Replication, Bypass RLS

postgres=# sudo -u postgres createdb -E UTF8 nominatim
postgres-# exit
Use \q to quit.
postgres-# \q
(nominatim-env) root@debian:~# nominatim import --osm-file /root/nominatim-project/Nominatim/osm-data/berlin-latest.osm.pbf --threads 4 --osm2pgsql-cache 2000
2024-09-14 12:33:16: Using project directory: /root
2024-09-14 12:33:17: Creating database
createdb: error: connection to server on socket "/var/run/postgresql/.s.PGSQL.5432" failed: FATAL:  role "root" does not exist
2024-09-14 12:33:17: FATAL: Creating new database failed.
(nominatim-env) root@debian:~# 

I checked local.php as well.everything looks good but database creation is failing

**Reply by mtmail** [reply]
Have you seen https://github.com/mediagis/nominatim-docker , or do you have requirements that are different from the defaults (they offer many configuration settings).

> I checked local.php as well.

Nominatim 4.x no longer uses that file. For Nominatim 4.4 you create a `.env` file in your project directoy. Then set the `NOMINATIM_DATABASE_DSN`. For example add a line `NOMINATIM_DATABASE_DSN=pgsql:user=nominatim;password=your_password;dbname=nominatim` in the `.env` file. The default is `pgsql:dbname=nominatim`
https://nominatim.org/release-docs/develop/customize/Settings/

**Reply by Shailesh913** [reply]
@mtmail Thank you for all your help.

It worked.Just a request if you can help me with Nginx instead of APache2. I tried to make changes in start.sh and later trying to troubleshoot but its giving me 502 gateway error.

Really grateful for all your help .
